### PR TITLE
Fixup comment typos and some voice logic

### PR DESCRIPTION
**Pull request by mobiuscog**
Just some comment corrections (Please tell me if you don't want that sort of thing - I avoided grammar and just changes anything that seemed like an obvious typo).

Also, some of the voice-check logic seemed backwards. I may have made a huge mistake, but hopefully my brain was working.

### CODE REVIEW

File: src/clap-saw-demo.cpp [hunk 1]
```diff
@@ -744,7 +750,7 @@ void ClapSawDemo::handleNoteOff(int port_index, int channel, int n)
 {
     for (auto &v : voices)
     {
-        if (v.state != SawDemoVoice::OFF && v.key == n && v.portid == port_index &&
+        if (v.state != SawDemoVoice::OFF && v.state != SawDemoVoice::NEWLY_OFF && v.key == n && v.portid == port_index &&
```
Comment (baconpaul) [inline src/clap-saw-demo.cpp:753]: This one is a real edge case and it’s fine. When I do a sweep tonight I might add a little voice member is active to clean up all these if logics!

File: src/saw-voice.cpp [hunk 2]
```diff
@@ -107,7 +107,7 @@ void SawDemoVoice::step()
     for (int i = 0; i < unison; ++i)
     {
         /*
-         * Use a cubic integrated saw and second deriv it at
+         * Use a cubic integrated saw and second derive it at
```
Comment (baconpaul) [inline src/saw-voice.cpp:110]: It actually should be deriv as in “take the second deriv” her

**Comment by baconpaul** [comment]
Graat thank you! 

I’ll merge this even though there’s one “e” I wouldn’t have added :)

File: src/saw-voice.cpp (hunk 2 above)
Comment (mobiuscog) [reply src/saw-voice.cpp:110]: As in short for derivation ? You can tell I'm not a maths guy 😆

Comment (baconpaul) [reply src/saw-voice.cpp:110]: Yeah that’s how the algo works, it uses the second derivative of the second integral of a sawtooth to suppress aliasing.

**Pull request merged**